FaceID/faceID.py

```diff
@@ -37,6 +41,205 @@
+
+
+# Handler function for the keypad
+def handleKeyPress(key):
+
+    global acceptInput
+    global code
+
+    # Add the pressed key to the code sequence when
+    # input is being accepted
+    if acceptInput and key != "#":
+        print(key)
+        code += key
+
+    # Stop accepting input when the end key is pressed
+    elif acceptInput and key == "#":
+        acceptInput = False
+
+keypad.registerKeyPressHandler(handleKeyPress)
+
+
+# Retrieve input from the keypad
+def getInput(sec):
+
+    global acceptInput
+
+    acceptInput = True
+    endTime = time.time() + sec
+
+    # Allow input to be accepted for sec seconds and
+    # stop when the end key is pressed
+    while acceptInput and time.time() < endTime:
+        pass
+    acceptInput = False
+
+    # Return whether the loop ended because time ran out
+    return (time.time() > endTime)
+
+
+# Retrieves and verifies the keycode
+def getKeycode(response):
+
+    global mylcd
+    global code
+
+    # Retrieve the employee's actual keycode
+    employee = response.json()
+    actualKeycode = employee["keycode"]
+
+    attempts = 0
+    enteredKeycode = -1
+    timeOut = False
+
+    # Accept keycode attempts while the most recently entered one is incorrect,
+    # there have been less than 3 attempts, and the input has not timed out
+    while str(actualKeycode) != enteredKeycode and attempts < 3 and not timeOut:
+
+        attempts += 1
+        mylcd.lcd_display_string("Please enter", 1, 2)
+        mylcd.lcd_display_string("keycode", 2, 4)
+
+        # Allow keycode to be entered and determine whether the input timed out
+        timeOut = getInput(10)
+        enteredKeycode = code
+        code = ""
+        mylcd.lcd_clear()
+
+        # Display if the entered keycode was incorrect
+        if str(actualKeycode) != enteredKeycode and not timeOut:
+            mylcd.lcd_display_string("Incorrect", 1, 3)
+            mylcd.lcd_display_string("keycode", 2, 4)
+            time.sleep(1.2)
+            mylcd.lcd_clear()
+
+    # Display if the input timed out and return error value
+    if timeOut:
+        mylcd.lcd_display_string("Input has", 1, 3)
+        mylcd.lcd_display_string("timed out", 2, 3)
+        time.sleep(1.2)
+        mylcd.lcd_clear()
+        return -1
+    # Display welcome message and return verified emp_ID
+    elif str(actualKeycode) == enteredKeycode:
+        first_name = employee["first_name"]
+        mylcd.lcd_display_string("Welcome", 1, 4)
+        mylcd.lcd_display_string(first_name, 2, (16 - len(first_name)) // 2)
+        return employee["emp_ID"]
+    # Display that there were too many failed attempts and return error value
+    else:
+        mylcd.lcd_display_string("Too many", 1, 4)
+        mylcd.lcd_display_string("failed attempts", 2)
+        time.sleep(1.2)
+        mylcd.lcd_clear()
+        return -1
+
+
+# Retrieves and verifies the emp_ID
+def getEmpID():
+
+    global mylcd
+    global code
+    global url
+    global headers
```

Comment:
I'm not super stoked about having all these global variables, but for the sake of time, it works. Just for the future, avoid global variables as much as you can :)